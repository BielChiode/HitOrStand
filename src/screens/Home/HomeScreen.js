import { Button, Image, StyleSheet, Text, View } from 'react-native'
import logo from '../../../assets/logo.png'

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image style={styles.image} source={logo} />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          style={styles.buttonStart}
          onPress={() => navigation.navigate('Game')}
          title="Iniciar jogo"
          color="#99CFBD"
          accessibilityLabel="Iniciar jogo"
        />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  imageContainer: {
    width: '100%',
    height: '60%',
    justifyContent: 'center',
    alignItems: 'center',
    // borderWidth: 2
  },
  image: {
    width: 300,
    height: 300
  },
  buttonContainer: {
    flex: 1,
    width: 250,
    height: 200,
    justifyContent: 'center',
    alignSelf: 'center'
  }
})
