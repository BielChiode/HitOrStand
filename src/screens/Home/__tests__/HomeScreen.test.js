import { fireEvent, render } from '@testing-library/react-native'
import HomeScreen from '../HomeScreen'

describe('Home', () => {
  const navigationMock = { navigate: jest.fn() }
  afterEach(() => jest.resetAllMocks())
  it('Deveria renderizar corretamente', () => {
    const { toJSON } = render(<HomeScreen navigation={navigationMock} />)
    expect(toJSON).toMatchSnapshot()
  })
  it('Deveria levar para a pÃ¡gina do jogo', () => {
    const { getByTestId } = render(<HomeScreen navigation={navigationMock} />)
    fireEvent.press(getByTestId('inicia-jogo'))
    expect(navigationMock.navigate).toHaveBeenCalledWith('Game')
  })
})
