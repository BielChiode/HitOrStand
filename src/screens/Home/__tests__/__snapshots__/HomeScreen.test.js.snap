// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Deveria renderizar corretamente 1`] = `[Function]`;
